<html>
<head>
<title>ArithmeticExceptionExample.java</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<style type="text/css">
.s0 { color: #cf8e6d;}
.s1 { color: #bcbec4;}
.s2 { color: #bcbec4;}
.s3 { color: #7a7e85;}
.s4 { color: #2aacb8;}
.s5 { color: #6aab73;}
</style>
</head>
<body bgcolor="#1e1f22">
<table CELLSPACING=0 CELLPADDING=5 COLS=1 WIDTH="100%" BGCOLOR="#606060" >
<tr><td><center>
<font face="Arial, Helvetica" color="#000000">
ArithmeticExceptionExample.java</font>
</center></td></tr></table>
<pre><span class="s0">public class </span><span class="s1">ArithmeticExceptionExample </span><span class="s2">{</span>
    <span class="s0">public static void </span><span class="s1">main</span><span class="s2">(</span><span class="s1">String</span><span class="s2">[] </span><span class="s1">args</span><span class="s2">) {</span>
        <span class="s3">// Two integers for division</span>
        <span class="s0">int </span><span class="s1">numerator </span><span class="s2">= </span><span class="s4">10</span><span class="s2">;</span>
        <span class="s0">int </span><span class="s1">denominator </span><span class="s2">= </span><span class="s4">0</span><span class="s2">; </span><span class="s3">// Setting denominator to zero to trigger the exception</span>

        <span class="s0">try </span><span class="s2">{</span>
            <span class="s3">// Attempt to perform division</span>
            <span class="s0">int </span><span class="s1">result </span><span class="s2">= </span><span class="s1">numerator </span><span class="s2">/ </span><span class="s1">denominator</span><span class="s2">; </span><span class="s3">// This will throw ArithmeticException</span>
            <span class="s1">System</span><span class="s2">.</span><span class="s1">out</span><span class="s2">.</span><span class="s1">println</span><span class="s2">(</span><span class="s5">&quot;Result: &quot; </span><span class="s2">+ </span><span class="s1">result</span><span class="s2">);</span>
        <span class="s2">} </span><span class="s0">catch </span><span class="s2">(</span><span class="s1">ArithmeticException e</span><span class="s2">) {</span>
            <span class="s3">// Handle the ArithmeticException</span>
            <span class="s3">// This block will be executed if there is an arithmetic error (e.g., division by zero)</span>
            <span class="s1">System</span><span class="s2">.</span><span class="s1">err</span><span class="s2">.</span><span class="s1">println</span><span class="s2">(</span><span class="s5">&quot;ArithmeticException occurred: &quot; </span><span class="s2">+ </span><span class="s1">e</span><span class="s2">.</span><span class="s1">getMessage</span><span class="s2">());</span>
        <span class="s2">} </span><span class="s0">catch </span><span class="s2">(</span><span class="s1">Exception e</span><span class="s2">) {</span>
            <span class="s3">// Handle any other exceptions that may occur</span>
            <span class="s1">System</span><span class="s2">.</span><span class="s1">err</span><span class="s2">.</span><span class="s1">println</span><span class="s2">(</span><span class="s5">&quot;An unexpected error occurred: &quot; </span><span class="s2">+ </span><span class="s1">e</span><span class="s2">.</span><span class="s1">getMessage</span><span class="s2">());</span>
        <span class="s2">}</span>
    <span class="s2">}</span>
<span class="s2">}</span></pre>
</body>
</html>